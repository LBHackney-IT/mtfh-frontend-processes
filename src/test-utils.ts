import {
  getAssetV1,
  getPersonV1,
  getProcessV1,
  getTenureV1,
  mockProcessV1,
  patchProcessV1,
  postProcessV1,
  server,
} from "@hackney/mtfh-test-utils";

beforeEach(() => {
  server.use(getTenureV1());
  server.use(getPersonV1());
  server.use(getAssetV1());
  server.use(postProcessV1());
  server.use(getProcessV1());
  server.use(patchProcessV1());
});

export const mockProcessSelectTenants = {
  ...mockProcessV1,
  currentState: { ...mockProcessV1.currentState, state: "SelectTenants" },
};

export const mockProcessAutomatedChecksFailed = {
  ...mockProcessV1,
  currentState: { ...mockProcessV1.currentState, state: "AutomatedChecksFailed" },
};

export const mockProcessAutomatedChecksPassed = {
  ...mockProcessV1,
  currentState: { ...mockProcessV1.currentState, state: "AutomatedChecksPassed" },
};

export const mockProcessInvalidState = {
  ...mockProcessV1,
  currentState: { ...mockProcessV1.currentState, state: "InvalidState" },
};

export const mockManualChecksPassedState = {
  ...mockProcessV1,
  currentState: { ...mockProcessV1.currentState, state: "ManualChecksPassed" },
};

export const mockManualChecksFailedState = {
  ...mockProcessV1,
  currentState: { ...mockProcessV1.currentState, state: "ManualChecksFailed" },
};

export const mockBreachChecksFailedState = {
  ...mockProcessV1,
  currentState: { ...mockProcessV1.currentState, state: "BreachChecksFailed" },
};

export const mockBreachChecksPassedState = {
  ...mockProcessV1,
  currentState: { ...mockProcessV1.currentState, state: "BreachChecksPassed" },
};

export const mockDocumentsRequestedDes = {
  ...mockProcessV1,
  currentState: { ...mockProcessV1.currentState, state: "DocumentsRequestedDes" },
};

export const mockProcessClosedState = {
  ...mockProcessV1,
  currentState: { ...mockProcessV1.currentState, state: "ProcessClosed" },
};

export const mockDocumentsRequestedDesAppointment = {
  ...mockProcessV1,
  currentState: {
    ...mockProcessV1.currentState,
    state: "DocumentsRequestedAppointment",
    processData: {
      formData: {
        appointmentDateTime: "2022-10-12T08:59:00.000Z",
      },
      documents: [],
    },
  },
  previousStates: [{ ...mockProcessV1.currentState, state: "DocumentsRequestedDes" }],
};

export const mockDocumentsRequestedAppointment = ({
  appointmentDateTime,
}: {
  appointmentDateTime: string;
}) => {
  return {
    ...mockProcessV1,
    currentState: {
      ...mockProcessV1.currentState,
      state: "DocumentsRequestedAppointment",
      processData: {
        formData: {
          appointmentDateTime,
        },
        documents: [],
      },
    },
  };
};

export const mockDocumentsAppointmentRescheduled = {
  ...mockProcessV1,
  currentState: {
    ...mockProcessV1.currentState,
    state: "DocumentsAppointmentRescheduled",
  },
};
