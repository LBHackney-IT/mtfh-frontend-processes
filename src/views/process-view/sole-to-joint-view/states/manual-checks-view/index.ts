export * from "./manual-checks-view";
