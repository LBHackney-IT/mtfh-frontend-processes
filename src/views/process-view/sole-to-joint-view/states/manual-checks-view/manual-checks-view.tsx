import { locale } from "../../../../../services";
import { AutomatedChecksPassedBox } from "../shared";

import { Box, Heading, List, StatusHeading, Text } from "@mtfh/common/lib/components";

const { views } = locale;
const { checkEligibility } = views;

export const ManualChecksFailedView = ({ process }): JSX.Element => {
  return (
    <div data-testid={`soletojoint-${process.currentState.state}`}>
      <Text>{checkEligibility.autoCheckIntro}</Text>
      <Heading variant="h4">{checkEligibility.autoCheckInfo}</Heading>
      <AutomatedChecksPassedBox />
      <Box variant="warning">
        <StatusHeading variant="warning" title={checkEligibility.failedChecks} />{" "}
        <div style={{ marginLeft: "60px" }}>
          <Text>
            All criteria must be passed in order for the applicant to be eligible. <br />
            Applicant has failed one or more of the further eligibility questions:
          </Text>
          <Heading variant="h4">Failed further eligibility question:</Heading>
          <List variant="bullets">
            <Text>
              The tenant and proposed tenant have not been living together for 12 months
              or more, or are not married or in a civil partnership
            </Text>
            <Text>
              The tenant or proposed tenant intends to occupy another property as their
              only or main home
            </Text>
            <Text>Then tenant is a survivor of a joint tenancy</Text>
            <Text>
              The proposed tenant has been evicted by London Borough of Hackney or another
              local authority or housing association
            </Text>
            <Text>
              The proposed tenant is subject to immigration control under the Asylum And
              Immigration Act 1996
            </Text>
            <Text>The tenant has a live notice seeking possession</Text>
            <Text>The tenant has rent arrears over Â£500</Text>
            <Text>Proposed joint tenant holds a tenancy or property elsewhere</Text>
          </List>
        </div>
      </Box>
    </div>
  );
};
