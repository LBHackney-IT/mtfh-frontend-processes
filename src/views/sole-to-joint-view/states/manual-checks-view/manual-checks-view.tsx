import { SoleToJointHeader } from "../../../../components";
import { locale } from "../../../../services";
import { CloseProcessView } from "../../shared/close-process-view";
import { AutomatedChecksPassedBox } from "../shared";

import { Box, Heading, List, StatusHeading, Text } from "@mtfh/common/lib/components";

const { views } = locale;
const { checkEligibility } = views;

export const ManualChecksFailedView = ({
  process,
  processConfig,
  mutate,
}): JSX.Element => {
  return (
    <div data-testid={`soletojoint-${process.currentState.state}`}>
      <SoleToJointHeader processConfig={processConfig} process={process} />
      <Text>{checkEligibility.autoCheckIntro}</Text>
      <Heading variant="h5">{checkEligibility.autoCheckInfo}</Heading>
      <AutomatedChecksPassedBox />
      <Box variant="warning">
        <StatusHeading variant="warning" title={checkEligibility.failedChecks} />{" "}
        <div style={{ marginLeft: "60px" }}>
          <Text size="sm">
            All criteria must be passed in order for the applicant to be eligible. <br />
            Applicant has failed one or more of the further eligibility questions:
          </Text>
          <Heading variant="h5">Failed further eligibility question:</Heading>
          <List variant="bullets">
            <Text size="sm">
              The tenant and proposed tenant have not been living together for 12 months
              or more, or are not married or in a civil partnership
            </Text>
            <Text size="sm">
              The tenant or proposed tenant intends to occupy another property as their
              only or main home
            </Text>
            <Text size="sm">Then tenant is a survivor of a joint tenancy</Text>
            <Text size="sm">
              The proposed tenant has been evicted by London Borough of Hackney or another
              local authority or housing association
            </Text>
            <Text size="sm">
              The proposed tenant is subject to immigration control under the Asylum And
              Immigration Act 1996
            </Text>
            <Text size="sm">The tenant has a live notice seeking possession</Text>
            <Text size="sm">The tenant has rent arrears over Â£500</Text>
          </List>
        </div>
      </Box>
      <CloseProcessView process={process} processConfig={processConfig} mutate={mutate} />
    </div>
  );
};
