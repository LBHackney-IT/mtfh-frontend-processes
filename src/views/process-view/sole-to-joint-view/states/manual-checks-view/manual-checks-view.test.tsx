import React from "react";

import { render } from "@hackney/mtfh-test-utils";

import { mockManualChecksFailedState } from "../../../../../test-utils";
import { ManualChecksFailedView } from "./manual-checks-view";

describe("manual-checks-view", () => {
  beforeEach(() => {
    jest.resetModules();
  });

  test("it renders ManualChecksView correctly", async () => {
    const { container } = render(
      <ManualChecksFailedView process={mockManualChecksFailedState} />,
      {
        url: "/processes/soletojoint/e63e68c7-84b0-3a48-b450-896e2c3d7735",
        path: "/processes/soletojoint/:processId",
      },
    );
    expect(container).toMatchSnapshot();
  });
});
