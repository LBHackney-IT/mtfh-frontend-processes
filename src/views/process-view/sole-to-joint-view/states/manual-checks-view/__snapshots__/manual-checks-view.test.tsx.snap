// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`manual-checks-view it renders ManualChecksView correctly 1`] = `
<div>
  <div
    data-testid="soletojoint-ManualChecksFailed"
  >
    <p
      class="lbh-body-m"
    >
      Manage My Home has used tenure, person and property information to automatically check eligibility for this application.
    </p>
    <h5
      class="lbh-heading-h5"
    >
      This is an automated check based on the data the system has. At this stage, the system does not have all the data required to make a decision, so these results are for guidance only and do not reflect accurate information.
    </h5>
    <div
      class="mtfh-box mtfh-box--success"
    >
      <h4
        class="lbh-heading-h4"
        style="margin-bottom: 0.5em;"
      >
        Passed automatic eligibility checks
      </h4>
      <div
        style="display: flex; align-items: center; margin: 0px 0px 0px -7px;"
      >
        <svg
          fill="none"
          height="32"
          viewBox="0 0 45 45"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M35 15.5127L18.775 33L11 24.6201L14.2591 21.1074L18.775 25.9746L31.7409 12L35 15.5127Z"
            fill="#00664F"
            fill-rule="evenodd"
          />
        </svg>
        <p
          class="lbh-body-s"
          style="margin: 0px;"
        >
          Applicant is a named tenure holder on the tenure
        </p>
      </div>
      <div
        style="display: flex; align-items: center; margin: 0px 0px 0px -7px;"
      >
        <svg
          fill="none"
          height="32"
          viewBox="0 0 45 45"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M35 15.5127L18.775 33L11 24.6201L14.2591 21.1074L18.775 25.9746L31.7409 12L35 15.5127Z"
            fill="#00664F"
            fill-rule="evenodd"
          />
        </svg>
        <p
          class="lbh-body-s"
          style="margin: 0px;"
        >
          Applicant is currently a sole tenant
        </p>
      </div>
      <div
        style="display: flex; align-items: center; margin: 0px 0px 0px -7px;"
      >
        <svg
          fill="none"
          height="32"
          viewBox="0 0 45 45"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M35 15.5127L18.775 33L11 24.6201L14.2591 21.1074L18.775 25.9746L31.7409 12L35 15.5127Z"
            fill="#00664F"
            fill-rule="evenodd"
          />
        </svg>
        <p
          class="lbh-body-s"
          style="margin: 0px;"
        >
          Secure tenures can be changed from a sole to joint tenancy
        </p>
      </div>
      <div
        style="display: flex; align-items: center; margin: 0px 0px 0px -7px;"
      >
        <svg
          fill="none"
          height="32"
          viewBox="0 0 45 45"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M35 15.5127L18.775 33L11 24.6201L14.2591 21.1074L18.775 25.9746L31.7409 12L35 15.5127Z"
            fill="#00664F"
            fill-rule="evenodd"
          />
        </svg>
        <p
          class="lbh-body-s"
          style="margin: 0px;"
        >
          Tenant's tenure is active
        </p>
      </div>
      <div
        style="display: flex; align-items: center; margin: 0px 0px 0px -7px;"
      >
        <svg
          fill="none"
          height="32"
          viewBox="0 0 45 45"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M35 15.5127L18.775 33L11 24.6201L14.2591 21.1074L18.775 25.9746L31.7409 12L35 15.5127Z"
            fill="#00664F"
            fill-rule="evenodd"
          />
        </svg>
        <p
          class="lbh-body-s"
          style="margin: 0px;"
        >
          The proposed tenant is over 18 years of age
        </p>
      </div>
      <div
        style="display: flex; align-items: center; margin: 0px 0px 0px -7px;"
      >
        <svg
          fill="none"
          height="32"
          viewBox="0 0 45 45"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M35 15.5127L18.775 33L11 24.6201L14.2591 21.1074L18.775 25.9746L31.7409 12L35 15.5127Z"
            fill="#00664F"
            fill-rule="evenodd"
          />
        </svg>
        <p
          class="lbh-body-s"
          style="margin: 0px;"
        >
          Proposed tenant is not a tenure holder or household member within the London Borough of Hackney
        </p>
      </div>
    </div>
    <div
      class="mtfh-box mtfh-box--warning"
    >
      <div
        class="mtfh-status-heading"
      >
        <svg
          fill="none"
          height="45"
          viewBox="0 0 45 45"
          width="45"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="22.5"
            cy="22.5"
            fill="white"
            r="21.75"
            stroke="#BE3A34"
            stroke-width="1.5"
          />
          <path
            d="M24.2969 26.6587H20.6035L19.8315 11.8701H25.0688L24.2969 26.6587ZM19.771 31.8354C19.771 30.9878 19.998 30.347 20.4521 29.9131C20.9062 29.4792 21.5672 29.2622 22.4351 29.2622C23.2726 29.2622 23.9185 29.4842 24.3726 29.9282C24.8368 30.3722 25.0688 31.008 25.0688 31.8354C25.0688 32.6326 24.8368 33.2633 24.3726 33.7275C23.9084 34.1816 23.2625 34.4087 22.4351 34.4087C21.5874 34.4087 20.9315 34.1867 20.4673 33.7427C20.0031 33.2886 19.771 32.6528 19.771 31.8354Z"
            fill="#BE3A34"
          />
        </svg>
        <h4
          class="lbh-heading-h4 mtfh-status-heading__title"
        >
          Not eligible for a sole to joint tenure
        </h4>
      </div>
       
      <div
        style="margin-left: 60px;"
      >
        <p
          class="lbh-body-s"
        >
          All criteria must be passed in order for the applicant to be eligible. 
          <br />
          Applicant has failed one or more of the further eligibility questions:
        </p>
        <h5
          class="lbh-heading-h5"
        >
          Failed further eligibility question:
        </h5>
        <ul
          class="lbh-list lbh-list--bullet"
        >
          <li>
            <p
              class="lbh-body-s"
            >
              The tenant and proposed tenant have not been living together for 12 months or more, or are not married or in a civil partnership
            </p>
          </li>
          <li>
            <p
              class="lbh-body-s"
            >
              The tenant or proposed tenant intends to occupy another property as their only or main home
            </p>
          </li>
          <li>
            <p
              class="lbh-body-s"
            >
              Then tenant is a survivor of a joint tenancy
            </p>
          </li>
          <li>
            <p
              class="lbh-body-s"
            >
              The proposed tenant has been evicted by London Borough of Hackney or another local authority or housing association
            </p>
          </li>
          <li>
            <p
              class="lbh-body-s"
            >
              The proposed tenant is subject to immigration control under the Asylum And Immigration Act 1996
            </p>
          </li>
          <li>
            <p
              class="lbh-body-s"
            >
              The tenant has a live notice seeking possession
            </p>
          </li>
          <li>
            <p
              class="lbh-body-s"
            >
              The tenant has rent arrears over Â£500
            </p>
          </li>
          <li>
            <p
              class="lbh-body-s"
            >
              Proposed joint tenant holds a tenancy or property elsewhere
            </p>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
